src/intermediate/IntermediateInstruction.h

```diff
@@ -505,8 +512,16 @@ namespace vc4c
 		using InstructionsIterator = FastModificationList<std::unique_ptr<IntermediateInstruction>>::iterator;
 		using ConstInstructionsIterator = FastModificationList<std::unique_ptr<IntermediateInstruction>>::const_iterator;
 	} // namespace intermediate
+
+
+
+  template<>
+  struct hash<intermediate::IntermediateInstruction> : public std::hash<std::string>
```

Comment:
Same here, I don't see a reason to hash an instruction directly.

If you must, you could also use the pointers.

Reply:
Same as the upper case. Removed